frontend/src/views/Announcements/AnnouncementsList.tsx

```diff
@@ -0,0 +1,213 @@
+import {
+  Button,
+  Icon,
+  IconButton,
+  Stack,
+  Toolbar,
+  Tooltip,
+  Typography,
+} from '@mui/material';
+import {
+  AnnouncementCreateRequestDto,
+  AnnouncementCreateResponseDto,
+  AnnouncementDeleteResponseDto,
+  AnnouncementFindAllResponseDto,
+  AnnouncementUpdateRequestDto,
+  AnnouncementUpdateResponseDto,
+  DtoAnnouncement,
+} from '@osk/shared';
+import { Box, Flex } from 'reflexbox';
+import useSWR from 'swr';
+import { UserRole } from '@osk/shared/src/types/user.types';
+import { useState } from 'react';
+import { FullPageLoading } from '../../components/FullPageLoading/FullPageLoading';
+import { GeneralAPIError } from '../../components/GeneralAPIError/GeneralAPIError';
+import { useActionModal } from '../../hooks/useActionModal/useActionModal';
+import { AnnouncementCard } from './AnnouncementCard/AnnouncementCard';
+import { useAuth } from '../../hooks/useAuth/useAuth';
+import { ActionModal } from '../../components/ActionModal/ActionModal';
+import { useCommonSnackbars } from '../../hooks/useCommonSnackbars/useCommonSnackbars';
+import { useMakeRequestWithAuth } from '../../hooks/useMakeRequestWithAuth/useMakeRequestWithAuth';
+import { AnnouncementFormModal } from './AnnouncementFormModal/AnnouncementFormModal';
+import { AnnouncementFormSchema } from './AnnouncementFormModal/AnnouncementFormModal.schema';
+
+export const AnnouncementsList = () => {
+  const {
+    data: announcementsData,
+    error: announcementsError,
+    mutate,
+  } = useSWR<AnnouncementFindAllResponseDto>('/api/announcements');
+  const pageTitle = 'Ogłoszenia';
+  const { role } = useAuth();
+  const { showErrorSnackbar, showSuccessSnackbar } = useCommonSnackbars();
+  const makeRequest = useMakeRequestWithAuth();
+  const {
+    setLoading: setDeleteModalLoading,
+    isLoading: isDeleteModalLoading,
+    isOpen: isDeleteModalOpen,
+    openModal: openDeleteModal,
+    closeModal: closeDeleteModal,
+  } = useActionModal();
+
+  const [announcementToBeDeleted, setAnnouncementToBeDeleted] =
+    useState<number>();
+  const [announcementToBeEdited, setAnnouncementToBeEdited] =
```

Comment:
Let's change it to `announcementIdToBeEdited`

Reply:
It's not Id though - for edit I store whole object